File: sdk/java/reflection/ReflectionField.php
<?php

namespace java\reflection;


class ReflectionField
{
    /**
     * @return string
     */
    public function getName(): string {}

    /**
     * @return int
     */
    public function getModifiers(): int {}

    /**
     * @return ReflectionClass
     */
    public function getDeclaringClass(): ReflectionClass {}

    /**
     * @param bool $accessible
     */
    public function setAccessible(bool $accessible): void {}

    /**
     * @return ReflectionClass
     */
    public function getType(): ReflectionClass {}

    /**
     * @param ReflectionObject $object
     * @return ReflectionObject
     */
    public function get(ReflectionObject $object): ReflectionObject {}

    /**
     * @param ReflectionObject $object
     * @return bool
     */
    public function getBoolean(ReflectionObject $object): bool {}

    /**
     * @param ReflectionObject $object
     */
    public function getByte(ReflectionObject $object) {}

    /**
     * @param ReflectionObject $object
     */
    public function getChar(ReflectionObject $object) {}

    /**
     * @param ReflectionObject $object
     */
    public function getShort(ReflectionObject $object) {}

    /**
     * @param ReflectionObject $object
     */
    public function getInt(ReflectionObject $object) {}

    /**
     * @param ReflectionObject $object
     */
    public function getLong(ReflectionObject $object) {}

    /**
     * @param ReflectionObject $object
     */
    public function getDouble(ReflectionObject $object) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function set(ReflectionObject $object, $value) {}

    /**
     * @param ReflectionObject $object
     * @param bool $value
     */
    public function setBoolean(ReflectionObject $object, bool $value) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function setByte(ReflectionObject $object, $value) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function setChar(ReflectionObject $object, $value) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function setShort(ReflectionObject $object, $value) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function setInt(ReflectionObject $object, $value) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function setLong(ReflectionObject $object, $value) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function setFloat(ReflectionObject $object, $value) {}

    /**
     * @param ReflectionObject $object
     * @param $value
     */
    public function setDouble(ReflectionObject $object, $value) {}
}

File: src-jvm/main/java/org/venity/javareflection/classes/ReflectionField.java
package org.venity.javareflection.classes;

import org.venity.javareflection.JavaReflectionExtension;
import php.runtime.Memory;
import php.runtime.annotation.Reflection;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

import java.lang.reflect.Field;

@Reflection.Name("ReflectionField")
@Reflection.Namespace(JavaReflectionExtension.NS)
public class ReflectionField extends BaseWrapper<Field> {
    public ReflectionField(Environment env, Field wrappedObject) {
        super(env, wrappedObject);
    }

    public ReflectionField(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Reflection.Signature
    public String getName() {
        return __wrappedObject.getName();
    }

    @Reflection.Signature
    public int getModifiers() {
        return __wrappedObject.getModifiers();
    }

    @Reflection.Signature
    public ReflectionClass getDeclaringClass(Environment environment) {
        return new ReflectionClass(environment, __wrappedObject.getDeclaringClass());
    }

    @Reflection.Signature
    public void setAccessible(boolean accessible) {
        __wrappedObject.setAccessible(accessible);
    }

    @Reflection.Signature
    public ReflectionClass getType() {
        return new ReflectionClass(__env__, __wrappedObject.getType());
    }

    @Reflection.Signature
    public ReflectionObject get(ReflectionObject object) throws IllegalAccessException {
        return new ReflectionObject(__env__, __wrappedObject.get(object.getWrappedObject()));
    }

    @Reflection.Signature
    public boolean getBoolean(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getBoolean(object.getWrappedObject());
    }

    @Reflection.Signature
    public byte getByte(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getByte(object.getWrappedObject());
    }

    @Reflection.Signature
    public char getChar(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getChar(object.getWrappedObject());
    }

    @Reflection.Signature
    public short getShort(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getShort(object.getWrappedObject());
    }

    @Reflection.Signature
    public int getInt(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getInt(object.getWrappedObject());
    }

    @Reflection.Signature
    public long getLong(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getLong(object.getWrappedObject());
    }

    @Reflection.Signature
    public float getFloat(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getFloat(object.getWrappedObject());
    }

    @Reflection.Signature
    public double getDouble(ReflectionObject object) throws IllegalAccessException {
        return __wrappedObject.getDouble(object.getWrappedObject());
    }

    @Reflection.Signature
    public void set(ReflectionObject object, Memory value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), Memory.unwrap(__env__, value));
    }

    @Reflection.Signature
    public void setBoolean(ReflectionObject object, boolean value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }

    @Reflection.Signature
    public void setByte(ReflectionObject object, byte value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }

    @Reflection.Signature
    public void setChar(ReflectionObject object, char value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }

    @Reflection.Signature
    public void setShort(ReflectionObject object, short value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }

    @Reflection.Signature
    public void setInt(ReflectionObject object, int value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }

    @Reflection.Signature
    public void setLong(ReflectionObject object, long value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }

    @Reflection.Signature
    public void setFloat(ReflectionObject object, float value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }

    @Reflection.Signature
    public void setDouble(ReflectionObject object, double value) throws IllegalAccessException {
        __wrappedObject.set(object.getWrappedObject(), value);
    }
}
